numba/core/pythonapi.py

```diff
@@ -131,10 +131,27 @@ def read_const(self, index):
         """
         Look up constant number *index* inside the environment body.
         A borrowed reference is returned.
+
+        The returned LLVM value may have NULL value at runtime which indicates
+        an error at runtime.
         """
         assert index < len(self.env.consts)
 
-        return self.pyapi.list_getitem(self.env_body.consts, index)
+        builder = self.pyapi.builder
+        consts = self.env_body.consts
+        ret = cgutils.alloca_once(builder, self.pyapi.pyobj, zfill=True)
+        with builder.if_else(cgutils.is_not_null(builder, consts)) as \
+                (br_not_null, br_null):
+            with br_not_null:
+                getitem = self.pyapi.list_getitem(consts, index)
+                builder.store(getitem, ret)
+            with br_null:
+                # This can happen when the Environment is accidentally released
```

Comment:
```suggestion
                # This can happen when the Environment is accidentally released and has subsequently been garbage collected.
```

Reply:
done in https://github.com/numba/numba/pull/6141/commits/62d2a8b08942146d6dfe84efec2e36f8712527fa